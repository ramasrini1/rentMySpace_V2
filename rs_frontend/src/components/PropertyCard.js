import React from "react";
import { Link } from "react-router-dom";
import './PropertyCard.css';

/** Show limited information about a property
 *
 * Is rendered by Home and MyListings to show a "card" for each property.
 *
 * Home -> PropertyCard
 * MyListings -> PropertyCard
 */
function PropertyCard({ propertyId, city, zip, cost, imgUrl, propertyType, btnName, btnLink }) {
  

  var myStyle = {
    fontSize: '15px',
    color: 'black'
  }
  return (
    <Link style={myStyle} to={`${btnLink}`}>
    <div className="card">
      <div className="card_body">
        { <img src={imgUrl} alt="house" className="card_image" /> }
        <h6 className="card_title">{city} <span style={myStyle}>{zip}</span></h6>
        <div className="card_description">Property Id: {propertyId} </div>
        <div className="card_description">Rental For: {propertyType} </div>
        <div className="card_description">Rate: ${cost} per hour</div>
      </div>
     
        <button className="card_btn">
          {btnName}
        </button>
     
    </div>
    </Link>
  
  );
}

export default PropertyCard;
