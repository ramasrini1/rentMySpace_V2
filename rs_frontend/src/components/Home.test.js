import React from "react";
import { render } from "@testing-library/react";
import Home from "./Home";
import PropertyCard from "./PropertyCard";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <PropertyCard
        />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("checks for listing", function () {
  const { getByText } = render(<Home />);
  let h = getByText("Property Listings:");
  expect(h).toHaveTextContent("Property Listings:");
});

// smoke test
it("renders without crashing", function() {
  render(<Home />);
});
