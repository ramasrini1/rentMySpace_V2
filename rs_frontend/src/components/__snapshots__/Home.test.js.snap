// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card_body"
    >
      <img
        alt="house"
        class="card_image"
      />
      <h6
        class="card_title"
      >
         
        <span
          style="font-size: 15px; color: blue;"
        />
      </h6>
      <div
        class="card_description"
      >
        Property Id:  
      </div>
      <div
        class="card_description"
      >
        Rental For:  
      </div>
      <div
        class="card_description"
      >
        Rate: $ per hour
      </div>
    </div>
    <a
      href="/undefined"
    >
      <button
        class="card_btn"
      />
    </a>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card_body"
    >
      <img
        alt="house"
        class="card_image"
      />
      <h6
        class="card_title"
      >
         
        <span
          style="font-size: 15px; color: blue;"
        />
      </h6>
      <div
        class="card_description"
      >
        Property Id:  
      </div>
      <div
        class="card_description"
      >
        Rental For:  
      </div>
      <div
        class="card_description"
      >
        Rate: $ per hour
      </div>
    </div>
    <a
      href="/undefined"
    >
      <button
        class="card_btn"
      />
    </a>
  </div>
</DocumentFragment>
`;
